shared/api/chat_template.cc

```diff
@@ -553,142 +544,148 @@ OrtxStatus TokenizerImpl::DeepSeekChatTemplate(std::string& output) {
+
+const std::unordered_map<std::string, std::string> model_to_template_map = {
+    // Phi-vision variants
+    { PHI_VISION_CHAT_TEMPLATE, "microsoft/Phi-3-vision-128k-instruct" },
+    { PHI_VISION_CHAT_TEMPLATE, "microsoft/Phi-3.5-vision-instruct" },
+
+    // Phi-3 variants
+    { PHI3_CHAT_TEMPLATE, "microsoft/Phi-3-mini-4k-instruct" },
+    { PHI3_CHAT_TEMPLATE, "microsoft/Phi-3-mini-128k-instruct" },
+
+    // Phi-3-5 variants
+    { PHI3_5_CHAT_TEMPLATE, "microsoft/Phi-3.5-mini-instruct" },
+    { PHI3_5_CHAT_TEMPLATE, "microsoft/Phi-3.5-MoE-instruct" },
+
+    // Phi-3-small variants
+    { PHI3_SMALL_CHAT_TEMPLATE, "microsoft/Phi-3-small-4k-instruct" },
+    { PHI3_SMALL_CHAT_TEMPLATE, "microsoft/Phi-3-small-128k-instruct" },
+
+    // Phi-3-medium variants
+    { PHI3_MEDIUM_CHAT_TEMPLATE, "microsoft/Phi-3-medium-4k-instruct" },
+    { PHI3_MEDIUM_CHAT_TEMPLATE, "microsoft/Phi-3-medium-128k-instruct" },
+
+    // Phi-4 variants
+    { PHI4_CHAT_TEMPLATE, "microsoft/Phi-4-multimodal-instruct" },
+
+    // Llama-2 variants
+    { LLAMA2_CHAT_TEMPLATE, "meta-llama/Llama-2-7b-chat-hf" },
+
+    // Llama-3 variants
+    { LLAMA3_CHAT_TEMPLATE, "meta-llama/Meta-Llama-3-8B-Instruct" },
+    { LLAMA3_CHAT_TEMPLATE, "meta-llama/Meta-Llama-3-70B-Instruct" },
+
+    // Llama-3-2 variants
+    { LLAMA3_2_CHAT_TEMPLATE, "meta-llama/Llama-3.2-1B-Instruct" },
+    { LLAMA3_2_CHAT_TEMPLATE, "meta-llama/Llama-3.2-3B-Instruct" },
+
+    // Llama-3-3 variants (Llama 3.1 & 3.3 share chat template logic)
+    { LLAMA3_3_CHAT_TEMPLATE, "meta-llama/Llama-3.1-8B-Instruct" },
+    { LLAMA3_3_CHAT_TEMPLATE, "meta-llama/Llama-3.1-70B-Instruct" },
+    { LLAMA3_3_CHAT_TEMPLATE, "meta-llama/Llama-3.3-70B-Instruct" },
+
+    // DeepSeek variants
+    { DEEPSEEK_CHAT_TEMPLATE, "deepseek-ai/DeepSeek-R1-Distill-Llama-70B" },
+    { DEEPSEEK_CHAT_TEMPLATE, "deepseek-ai/DeepSeek-R1-Distill-Qwen-32B" }
+};
+
 void TokenizerImpl::InitializeChatParameters(
-    bool add_prompt,
-    const std::string& eos,
-    const std::string& bos,
```

Comment:
can we still keep `eos` and `bos` here? and below where you set them to `tok_config_->bos_token_` and `tok_config_->eos_token_`, we can check if the parameters exist, if so we set to them, otherwise if the params are empty then we use the tok_config values? this way, for our own testing, we can just pass these vals in and avoid having to include the tokenizer files for each of the tokenizers we want to test (i will have a extensive unit testing PR soon that tests all the core functions for the different models on a number of different cases)

Reply:
trying the python unit test to do extensive tests. I cannot expect who can manually fill up all these arguments for many models.

And this InitializeChatParameters function breaks the thread-safe for tokenizer object. For example, when a tokenizer object was created but the template function was invoked in multiple thread, the messages and other member variables can be misused due to these functions not read-only (missing const function specifier).

Reply:
sounds good! will move my unit testing to python too and adjust based on these changes.